src/main/java/com/algorand/algosdk/v2/client/common/Client.java

```diff
@@ -0,0 +1,181 @@
+import com.algorand.algosdk.v2.client.algod.GetBlock;
+import com.algorand.algosdk.v2.client.algod.GetSupply;
+import com.algorand.algosdk.v2.client.algod.RegisterParticipationKeys;
+import com.algorand.algosdk.v2.client.algod.ShutdownNode;
+import com.algorand.algosdk.v2.client.algod.GetStatus;
+import com.algorand.algosdk.v2.client.algod.WaitForBlock;
+import com.algorand.algosdk.v2.client.algod.RawTransaction;
+import com.algorand.algosdk.v2.client.algod.TransactionParams;
+import com.algorand.algosdk.v2.client.algod.GetPendingTransactions;
+import com.algorand.algosdk.v2.client.algod.PendingTransactionInformation;
+import com.algorand.algosdk.v2.client.indexer.SearchForAccounts;
+import com.algorand.algosdk.v2.client.indexer.LookupAccountByID;
+import com.algorand.algosdk.v2.client.indexer.LookupAccountTransactions;
+import com.algorand.algosdk.v2.client.indexer.SearchForAssets;
+import com.algorand.algosdk.v2.client.indexer.LookupAssetByID;
+import com.algorand.algosdk.v2.client.indexer.LookupAssetBalances;
+import com.algorand.algosdk.v2.client.indexer.LookupAssetTransactions;
+import com.algorand.algosdk.v2.client.indexer.LookupBlock;
+import com.algorand.algosdk.v2.client.indexer.SearchForTransactions;
+
+
+public class Client {
+	
+	private OkHttpClient client;
+	private String host;
+	private int port;
+
+	public int getPort() {
+		return port;
+	}
+	
+	public String getHost() {
+		return host;
+	}
+	
+	public static HttpUrl getHttpUrl(QueryData qData, int port, String host) {
+		HttpUrl.Builder httpUrlBuilder = (new HttpUrl.Builder()).scheme("http").port(port).host(host);
+		for (String ps : qData.pathSegments) {
+			httpUrlBuilder.addPathSegment(ps);
+		}
+		for (Entry<String, String> kvp : qData.queries.entrySet()) {
+			httpUrlBuilder.addQueryParameter(kvp.getKey(), kvp.getValue());
+		}
+		HttpUrl httpUrl = httpUrlBuilder.build();
+		return httpUrl;
+	}
+	
+	public Client(String host, int port) {
+		MediaType.parse("application/json; charset=utf-8");
+		this.host = host;
+		this.port = port;
+		this.client = new OkHttpClient();
+	}
+	
+	public Response executeCall(QueryData qData, String getOrPost) throws Exception {
+		
+		HttpUrl httpUrl = getHttpUrl(qData, port, host);
+		Request request = new Request.Builder()
+				.url(httpUrl)
+				.get()
+				.build();
+/*		RequestBody body = RequestBody.create(mediaType, requestString);
+		Request request = new Request.Builder()
+				.url(httpUrl)
+				.post(body)
+				.build();*/
+		Response response = null;
+		try {
+			response = client.newCall(request).execute();	        
+		} catch (Exception e) {
+			throw e;
+		}
+		return response;
+		/*
+		String responseString = response.body().string();
+		if (response.code() != 200) {
+			throw new RuntimeException(responseString);
+		}
+		return responseString;
+        // String contentType = response.headers().get("Content-Type");
+*/
+	}
+	
+	
+	public AccountInformation AccountInformation(String address) {
```

Comment:
`Address`

Reply:
Also we should have comments on these functions

Reply:
Fixed.